src/libraries/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/HashProviderDispenser.Windows.cs

```diff
@@ -58,33 +69,77 @@ public static unsafe int HashData(string hashAlgorithmId, ReadOnlySpan<byte> sou
                     throw new CryptographicException();
                 }
 
-                if (!s_useCompatOneShot)
+                HashUpdateAndFinish(cachedAlgorithmHandle, hashSize, source, destination);
+
+                return hashSize;
+            }
+
+            private static unsafe bool HashDataUsingPseudoHandle(string hashAlgorithmId, ReadOnlySpan<byte> source, Span<byte> destination, out int hashSize)
+            {
+                hashSize = default;
+
+                Interop.BCrypt.BCryptAlgPseudoHandle algHandle;
+                int digestSizeInBytes;
```

Comment:
I wonder if pulling this hashSize logic up and completely avoid the call to `BCryptGetProperty` would gain any worthwhile wins even in the "slow" path.